### ISSUE TITLE
Unable to skip git submodule update

### ISSUE THREAD

**Issue by sjagoe** [ansible/ansible#9785]
##### Issue Type:

Feature Request
##### Ansible Version:

ansible 1.7.2
##### Environment:

Running: Ubuntu 14.04

Managing: Ubuntu 14.04
##### Summary:

I am attempting to keep a git repository up to date ignoring the
submodules. The recursive: false option to the git ansible module only
appears to have an effect in the initial clone. For any update, ansible
will always attempt to update the submodule with no (apparent) way to
disable this behaviour.
##### Steps To Reproduce:

Add an option to disable submodule updates entirely:

```
git:
  # Example option to disable submodule updates
  update_submodules: false
  recursive: false
  repo: "https://git.example.org/repo.git"
  dest: "{{ checkout }}"
  version: master
```
##### Expected Results:

On initial clone the submodules are ignored (due to `recursive: false`),
and on all further updates, the submodules remain ignored and not
updated.
##### Actual Results:

On initial clone the submodules are ignored (due to `recursive: false`),
and on all further updates, if the submodule ref has changed, the
following error is observed:

```
failed: [172.17.5.140] => {"failed": true}
msg: Failed to init/update submodules: Cloning into '<dest>'...
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Clone of 'user@example.org:<reponame>' into submodule path '<dest>' failed

FATAL: all hosts have already failed -- aborting
```

**Comment by sjagoe** [comment]
My current workaround is to ignore errors. I can't find any other way around this.

**Comment by sjagoe** [comment]
Hmm. This seems to be fixed in Ansible 1.8. Sorry for the noise.
